platforms/chibios/drivers/spi_master.c

```diff
@@ -267,17 +318,30 @@ spi_status_t spi_write(uint8_t data) {
 
 spi_status_t spi_read(void) {
     uint8_t data = 0;
+
+#if defined(QMK_MCU_SERIES_STM32H7XX)
+    SCB_InvalidateDCache_by_Addr((uint32_t *)(((uint32_t)&data) & ~(uint32_t)0x1F), 1 + 32);
+#endif
```

Comment:
I'm not sure whether `Invalidate` is the proper operation to do here, because the buffer is not aligned to a cache line, therefore we would be invalidating some unrelated areas.  Doing a `CleanInvalidate` instead should be safer, because the nearby memory may have modifications stored in the cache which need to be written back.

(Actually some ARM implementation may silently upgrade any `Invalidate` operations to `CleanInvalidate`, and that actually may cause difficulties when managing DMA buffers — doing `Invalidate` after the DMA operation does not work, instead a `CleanInvalidate` must be performed before the DMA operation.  Not sure how the relevant STM32 MCUs actually behave though, and depending on that upgrade won't be safe either.)